File: app/src/main/java/org/scp/gymlog/ui/training/ExerciseRows.java
package org.scp.gymlog.ui.training;

import org.scp.gymlog.model.Exercise;
import org.scp.gymlog.ui.training.rows.ITrainingRow;

import java.util.ArrayList;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ExerciseRows extends ArrayList<ITrainingRow> {
    private Exercise exercise;
    public ExerciseRows(Exercise exercise) {
        this.exercise = exercise;
    }
}

File: app/src/main/java/org/scp/gymlog/ui/training/ExerciseRows.kt
package org.scp.gymlog.ui.training

import org.scp.gymlog.model.Variation
import org.scp.gymlog.ui.training.rows.ITrainingRow

class ExerciseRows(
    variation: Variation,
    val superSet: Int? = null
): MutableList<ITrainingRow> by mutableListOf()
{
    private val _variations: MutableSet<Variation> = mutableSetOf(variation)

    val variations get() = _variations.toList()
    val exercise get() = _variations.single().exercise

    fun addVariation(variation: Variation) = _variations.add(variation)
}